internal/perms/grants.go

```diff
@@ -343,154 +393,219 @@ func Parse(scopeId, grantString string, opt ...Option) (Grant, error) {
 		}
 	}
 
-	opts := getOpts(opt...)
-
-	// Check for templated values ID, and substitute in with the authenticated
-	// values if so. If we are using a dummy user or account ID, store the
-	// original ID and return it at the end; this is usually the case when
-	// validating grant formats.
-	var origId string
-	if grant.id != "" && strings.HasPrefix(grant.id, "{{") {
-		id := strings.TrimSuffix(strings.TrimPrefix(grant.id, "{{"), "}}")
-		id = strings.TrimSpace(id)
-		switch id {
-		case "user.id", ".User.Id":
-			if opts.withUserId != "" {
-				grant.id = strings.ToValidUTF8(opts.withUserId, string(unicode.ReplacementChar))
-			}
-			// Otherwise, substitute in a dummy value
-			origId = grant.id
-			grant.id = "u_dummy"
-		case "account.id", ".Account.Id":
-			if opts.withAccountId != "" {
-				grant.id = strings.ToValidUTF8(opts.withAccountId, string(unicode.ReplacementChar))
-			}
-			origId = grant.id
-			grant.id = "acctoidc_dummy"
-		default:
-			return Grant{}, errors.NewDeprecated(errors.InvalidParameter, op, fmt.Sprintf("unknown template %q in grant %q value", grant.id, "id"))
-		}
+	if grant.id != "" && len(grant.ids) > 0 {
+		return Grant{}, errors.NewDeprecated(errors.InvalidParameter, op, fmt.Sprintf("input grant string %q contains both %q and %q fields", grantString, "id", "ids"))
 	}
-
-	if err := grant.validateType(); err != nil {
-		return Grant{}, errors.WrapDeprecated(err, op)
+	if len(grant.ids) > 1 && slices.Contains(grant.ids, "*") {
+		return Grant{}, errors.NewDeprecated(errors.InvalidParameter, op, fmt.Sprintf("input grant string %q contains both wildcard and non-wildcard values in %q field", grantString, "ids"))
 	}
 
-	if err := grant.parseAndValidateActions(); err != nil {
-		return Grant{}, errors.WrapDeprecated(err, op)
-	}
+	opts := getOpts(opt...)
```

Comment:
Should this also be changed to return an error? I notice you updated another options file for that

Reply:
No, I did that package specifically to solve some test issues I found locally when those env vars are set since they didn't act as true/false. At other points we can potentially update other options but that was for a specific reason at the moment.